crates/hyperqueue/src/client/resources.rs

```diff
@@ -79,85 +63,82 @@ mod test {
+            expect_parser_error(all_consuming(p_resource_request()), "=1"),
+            "Unexpected token found, expected something else:\n  =1\n  |\n  --- Unexpected token `=`\n",
         );
     }
 
     #[test]
     fn test_parse_identifier_slash() {
         assert_eq!(
             parse_resource_request("a/b=1").unwrap(),
-            ("a/b".to_string(), AllocationRequest::Compact(1))
+            (
+                "a/b".to_string(),
+                AllocationRequest::Compact(ResourceAmount::new_units(1))
+            )
         );
     }
 
     #[test]
     fn test_parse_identifier_start_with_digit() {
-        check_parse_error(
-            p_resource_request,
-            "1a=1",
-            r#"Parse error
-expected Resource identifier at character 0: "1a=1"
-expected Letter at character 0: "1a=1"
-  expected: Verify at character 0: "1a=1""#,
+        assert_eq!(
+            expect_parser_error(all_consuming(p_resource_request()), "1a=1"),
+            "Unexpected token found, expected something else:\n  1a=1\n  |\n  --- Unexpected token `1`\n",
         );
     }
 
     #[test]
     fn test_parse_zero_resources() {
-        check_parse_error(
-            p_resource_request,
-            "aa=0",
-            r#"Parse error
-expected Resource amount at character 3: "0"
-  "Requesting zero resources is not allowed" at character 3: "0""#,
+        assert_eq!(
+            expect_parser_error(all_consuming(p_resource_request()), "aa=0"),
+            "Unexpected end of input found, expected something else:\n  aa=0\n     |\n     --- Error: Zero resources cannot be requested\n",
         );
     }
 
     #[test]
     fn test_parse_resource_request_error() {
-        check_parse_error(
-            p_resource_request,
-            "",
-            r#"Parse error
-expected Resource identifier at the end of input
-expected Letter at the end of input
-  expected something at the end of input"#,
+        assert_eq!(
+            expect_parser_error(all_consuming(p_resource_request()), ""),
+            "Unexpected end of input found, expected something else:\n(the input was empty)",
         );
-        check_parse_error(
-            p_resource_request,
-            "a",
-            r#"Parse error
-expected "=" at the end of input"#,
+        assert_eq!(
+            expect_parser_error(all_consuming(p_resource_request()), "a"),
+            "Unexpected end of input found, expected =:\n  a\n   |\n   --- Unexpected end of input\n",
         );
-        check_parse_error(
-            p_resource_request,
-            "a=x",
-            "Parse error\nexpected Resource amount at character 2: \"x\"\n  expected one of the following 2 variants:\n    expected \"all\" at character 2: \"x\"\n    or\n    expected integer at character 2: \"x\"",
+        assert_eq!(
+            expect_parser_error(all_consuming(p_resource_request()), "a=x"),
+            "Unexpected token found, expected all:\n  a=x\n    |\n    --- Unexpected token `x`\n",
```

Comment:
The parse errors have regressed somewhat, as e.g. `ResourceAmount` isn't mentioned anymore. But let's not resolve that here.